### DISCUSSION TITLE
Inline environment variables

### DISCUSSION THREAD

**Discussion by pepicrft** [ElMassimo/vite_ruby#59] [Q&A] [answered]
Hi there again!

I'm trying to adjust the Vite process to inject environment variables into the served Javascript like the [babel-plugin-transform-inline-environment-variables](https://www.npmjs.com/package/babel-plugin-transform-inline-environment-variables) plugin does. However I couldn't find an API nor can think of a set up to achieve that with Vite.

Is there a way to achieve that?

**Comment by ElMassimo** [comment, marked as answer]
Hi Pedro!

[Vite uses dotenv](https://vitejs.dev/guide/env-and-mode.html#env-files) to load environment variables.

Only `VITE_` prefixed environment variables will be available in your client code, in `import.meta.env`.

See [this section in the Vite docs](https://vitejs.dev/guide/env-and-mode.html) for more information.

<hr/>

If using `VITE_` prefixed variables is not an option, you could also use [`define`](https://vitejs.dev/config/#define) to define constants in your `vite.config.ts`, which is a bit harder to use but [more flexible](https://github.com/vitejs/vite/blob/main/packages/playground/define/vite.config.js).

```js
  define: {
    'process.env.BUGSNAG_FRONTEND_KEY': JSON.stringify(process.env.BUGSNAG_FRONTEND_KEY),
  },
```

**Reply by pepicrft** [reply]
Thanks a lot @ElMassimo. I tried your suggestion and it works great 🚀

**Comment by ElMassimo** [comment]
[plugin]: https://github.com/ElMassimo/vite-plugin-environment

For @pepibumur and anyone reading this in the future–I've made a [plugin] to simplify this process, and allow providing defaults.

<hr/>

You can use <kbd>[vite-plugin-environment][plugin]</kbd> to expose environment variables to your client code:

```js
import { defineConfig } from 'vite'
import EnvironmentVariables from 'vite-plugin-environment'

export default defineConfig({
  plugins: [
    EnvironmentVariables(['BUGSNAG_API_KEY']),
  ],
})
```
